plugins/org.jkiss.dbeaver.ext.generic/src/org/jkiss/dbeaver/ext/generic/model/GenericExecutionContext.java

```diff
@@ -217,10 +217,10 @@ public boolean supportsCatalogChange() {
         GenericDataSource dataSource = getDataSource();
         if (dataSource.isSelectedEntityFromAPI() || !CommonUtils.isEmpty(dataSource.getQuerySetActiveDB())) {
             if (CommonUtils.isEmpty(dataSource.getSelectedEntityType())) {
-                return dataSource.hasCatalogs();
+                return dataSource.hasCatalogs() && !((GenericDataSourceInfo) dataSource.getInfo()).supportsCatalogSelection();
```

Comment:
Either extract this to a variable, or update `GenericDataSource#getInfo()` to return `GenericDataSourceInfo`.